File: ClientJava/src/main/java/Domain/AgeCategory.java
package Domain;

/**
 * Created by andrei on 2017-04-28.
 */
public class AgeCategory implements HasId<String> {
    private String name;
    private Integer minAge;
    private Integer maxAge;

    public AgeCategory() {
    }

    public AgeCategory(String name, Integer minAge, Integer maxAge) {
        this.name = name;
        this.minAge = minAge;
        this.maxAge = maxAge;
    }

    public String getId() {
        return name;
    }

    public void setId(String id) {
        name = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Integer getMinAge() {
        return minAge;
    }

    public void setMinAge(Integer minAge) {
        this.minAge = minAge;
    }

    public Integer getMaxAge() {
        return maxAge;
    }

    public void setMaxAge(Integer maxAge) {
        this.maxAge = maxAge;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof AgeCategory)) return false;

        AgeCategory that = (AgeCategory) o;

        if (getName() != null ? !getName().equals(that.getName()) : that.getName() != null) return false;
        if (getMinAge() != null ? !getMinAge().equals(that.getMinAge()) : that.getMinAge() != null) return false;
        return getMaxAge() != null ? getMaxAge().equals(that.getMaxAge()) : that.getMaxAge() == null;
    }

    @Override
    public int hashCode() {
        int result = getName() != null ? getName().hashCode() : 0;
        result = 31 * result + (getMinAge() != null ? getMinAge().hashCode() : 0);
        result = 31 * result + (getMaxAge() != null ? getMaxAge().hashCode() : 0);
        return result;
    }

    @Override
    public String toString() {
        return "AgeCategory{" +
                "name='" + name + '\'' +
                ", minAge=" + minAge +
                ", maxAge=" + maxAge +
                '}';
    }
}

File: ServerCSharp/ServerCSharp/Domain/AgeCategory.cs
using System;

namespace ServerCSharp.Domain
{
   public class AgeCategory : IHasId<String>
   {
      public String Name { get; set; }
      public Int32 MinAge { get; set; }
      public Int32 MaxAge { get; set; }
      public AgeCategory()
      {
      }

      public AgeCategory(String name, int minAge, int maxAge)
      {
         Name = name;
         MinAge = minAge;
         MaxAge = maxAge;
      }

      public void SetId(String name)
      {
         Name = name;
      }

      public String GetId()
      {
         return Name;
      }

      public override string ToString()
      {
         return $"{nameof(Name)}: {Name}, {nameof(MinAge)}: {MinAge}, {nameof(MaxAge)}: {MaxAge}";
      }

      protected bool Equals(AgeCategory other)
      {
         return Name == other.Name && MinAge == other.MinAge && MaxAge == other.MaxAge;
      }

      public override bool Equals(object obj)
      {
         if (ReferenceEquals(null, obj)) return false;
         if (ReferenceEquals(this, obj)) return true;
         if (obj.GetType() != this.GetType()) return false;
         return Equals((AgeCategory) obj);
      }

      public override int GetHashCode()
      {
         unchecked
         {
            var hashCode = Name.GetHashCode();
            hashCode = (hashCode * 397) ^ MinAge;
            hashCode = (hashCode * 397) ^ MaxAge;
            return hashCode;
         }
      }

      public static bool operator ==(AgeCategory left, AgeCategory right)
      {
         return Equals(left, right);
      }

      public static bool operator !=(AgeCategory left, AgeCategory right)
      {
         return !Equals(left, right);
      }
   }
}

